packages/contracts/contracts/registryManager/RegistryManager.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+import { ICommon } from "../interfaces/ICommon.sol";
+import { IOwnable } from "../interfaces/IOwnable.sol";
+import { IRecipientRegistry } from "../interfaces/IRecipientRegistry.sol";
+import { IRegistryManager } from "../interfaces/IRegistryManager.sol";
+
+/// @title RegistryManager
+/// @notice Contract that allows to use send, approve, reject requests to RecipientRegistry.
+contract RegistryManager is Ownable, IRegistryManager, ICommon {
+  /// @notice requests
+  mapping(uint256 => Request) internal requests;
+
+  /// @inheritdoc IRegistryManager
+  uint256 public requestCount;
+
+  /// @notice Initialize registry manager
+  constructor() payable Ownable(msg.sender) {}
+
+  /// @notice Check if request is valid
+  modifier isValidRequest(Request calldata request) {
+    if (request.registry == address(0)) {
+      revert ValidationError();
+    }
+
+    if (request.recipient.recipient == address(0)) {
+      revert ValidationError();
+    }
+
+    if (IOwnable(request.registry).owner() != address(this)) {
+      revert ValidationError();
+    }
+
+    uint256 count = IRecipientRegistry(request.registry).recipientCount();
+    bool withIndex = request.requestType == RequestType.Change || request.requestType == RequestType.Remove;
+
+    if (request.index >= count && withIndex) {
+      revert ValidationError();
+    }
+
+    _;
+  }
+
+  /// @notice Check if request is pending and exists
+  /// @param index Request index
+  modifier isPending(uint256 index) {
+    if (index >= requestCount || requests[index].status != Status.Pending) {
+      revert OperationError();
+    }
+
+    _;
+  }
+
+  /// @inheritdoc IRegistryManager
+  function process(Request calldata request) public virtual override isValidRequest(request) {
+    requests[requestCount] = request;
+    requests[requestCount].status = Status.Pending;
```

Comment:
would it be cheaper to set to memory and change before storing to storing to state? so we avoid another write to state